R/preprocess.R

```diff
@@ -106,6 +106,42 @@ enw_assign_group <- function(obs, by = c()) {
   return(obs = obs[])
 }
 
+#' @title FUNCTION_TITLE
+#' @description FUNCTION_DESCRIPTION
+#' @param obs PARAM_DESCRIPTION
+#' @return OUTPUT_DESCRIPTION
+#' @family preprocess
+#' @export
+#' @importFrom data.table as.data.table copy
+enw_add_delay <- function(obs) {
+  obs <- data.table::as.data.table(obs)
+  obs[, report_date := as.IDate(report_date)]
+  obs[, reference_date := as.IDate(reference_date)]
+  obs[, delay := as.numeric(report_date - reference_date)]
+  return(obs = obs[])
+}
+
+#' @title FUNCTION_TITLE
+#' @description FUNCTION_DESCRIPTION
+#' @param obs PARAM_DESCRIPTION
+#'
+#' @return OUTPUT_DESCRIPTION
+#'
+#' @family preprocess
+#' @export
+#' @importFrom data.table copy
+enw_add_max_reported <- function(obs) {
+  obs <- data.table::copy(obs)
```

Comment:
Do we have a clear strategy when to copy `obs`? In this case, e.g. `enw_latest_data` will also copy `obs` again, so it looks like a redundant copy to me.

Reply:
That is a good point. I think in general we don't ever want to depend on the input being a `data.table` and we also don't want to ever alter in place someone's input. For internal functions though we can do both those things. At the moment gone pretty maximal with exported functions. We can definitely pair those down as we stabilise the package. 

In terms of this copy you are right and this looks like a duplicate 😢